sqlmesh/core/model/definition.py

```diff
@@ -2634,6 +2637,64 @@ def _validate_model_fields(klass: t.Type[_Model], provided_fields: t.Set[str], p
         raise_config_error(f"Invalid extra fields {extra_fields} in the model definition", path)
 
 
+def _validate_model_properties(meta_fields: dict[str, t.Any], path: Path) -> None:
```

Comment:
Instead of doing it here, can we have a custom Pydantic validator for `session_properties_` in `model/meta.py` and remove the field from the generic property validator here: https://github.com/TobikoData/sqlmesh/blob/main/sqlmesh/core/model/common.py#L425-L432.

This way all model types will benefit from this and not just SQL models.